viewing_party/party.py

```diff
@@ -0,0 +1,193 @@
+    Otherwise, the function will simply return user_data. 
+    """
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+    return user_data
+
+# Wave 2
+def get_watched_avg_rating(user_data):
+    """
+    The function calculates and returns the average rating from 
+    a list of movies that the user has watched. The function 
+    returns 0 if the user's watched list is empty.
+    """
+    watched_movies = user_data["watched"]
+    sum = 0
+    if len(watched_movies) > 0:
+        for movie in watched_movies:
+            sum += movie["rating"]
+        avg_rating = sum/len(watched_movies)
+        return avg_rating
+    return 0
+
+def get_most_watched_genre(user_data):
+    """
+    The function returns the genre that is most frequently watched 
+    by the user from a list of movies that the user has watched. 
+    The function returns None if the list of movies that the user 
+    has watched is empty. 
+    """
+    genre_frequency = {}
+    if len(user_data["watched"]) == 0:
+        return None
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre not in genre_frequency:
+            genre_frequency[genre] = 1
+        else:
+            genre_frequency[genre] += 1
+
+    for genre, frequency in genre_frequency.items():
+        if frequency  == max(genre_frequency.values()):
+            return genre
+
+# # Wave 3
+# Helper function that will be used for waves 3-5
+def get_friends_watched(user_data):
+    """
+    Depending on the parameter, the function returns
+    different types of information about the movies that the
+    user's friends have watched. 
+    """
+    friends_watched = []
+    for watched_movies in user_data["friends"]:
+        for title in watched_movies["watched"]:
+            friends_watched.append(title)
+    return friends_watched
+
+def get_unique_watched(user_data):
+    """
+    This function returns a list of dictionaries that represents a 
+    list of movies that the user has watched, but none of their 
+    friends have watched. 
+    """
+    friends_watched = get_friends_watched(user_data)
+
+    unique_watched = []
+    for title in user_data["watched"]:
+        if title not in friends_watched:
+            if title not in unique_watched:
+                unique_watched.append(title)
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    """
+    This function returns a list of dictionaries that represents 
+    a list of movies that the user's friends have watched, but the 
+    user has not watched. 
+    """
+    friends_watched = get_friends_watched(user_data)
+
+    friends_unique_watched = []
+    for title in friends_watched:
+        if title not in user_data["watched"]:
+            if title not in friends_unique_watched:
+                friends_unique_watched.append(title)
```

Comment:
we can combine these two statements into one compound conditional statement:

```suggestion
        if title not in user_data["watched"] and title not in friends_unique_watched:
                friends_unique_watched.append(title)
```